pkg/reftracker/ref_tracker.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package reftracker
+
+import (
+	"fmt"
+	"strings"
+)
+
+var appExists struct{}
```

Comment:
since it's conventional to use `struct{}` as a sentinel value it's a bit weird to see as variable. in places where we use it i would recommend to do `struct{}{}`. (less indirection).

Reply:
Added